src/concurrent.rs

```diff
@@ -289,33 +300,58 @@ mod tests {
         assert_eq!(0, reader2.read(&mut buffer).expect("Expected 0 byte count as EOF after close"));
     }
 
-    #[test]
-    fn test_shareable_file_close_unblocks_reads_without_error() {
+    fn try_shareable_file_close_unblocks_reads_without_error() {
         let dir = tempfile::tempdir().unwrap();
         let path = dir.path().join("pipe");
         unistd::mkfifo(&path, sys::stat::Mode::S_IRUSR | sys::stat::Mode::S_IWUSR).unwrap();
 
-        // We need to write to the FIFO for the read-only open below to not block.
+        // We need to open the FIFO for writes so that the open for reads doesn't block.  But we
+        // have to do this open in a separate thread because it will block too.
+        let (writer_unblock_tx, writer_unblock_rx) = mpsc::channel();
         let writer_handle = {
             let path = path.clone();
             thread::spawn(move || {
-                let mut writer = fs::File::create(&path).unwrap();
-                write!(writer, "some text longer than we'll ever read")
+                let _keep_open = fs::File::create(path).unwrap();
```

Comment:
Hmm... I don't think I can open the writer on the main thread because it would block, but your comment gave me an idea -- or maybe that's what you were proposing after all. I can just return the result of create() here and handle any value in the parent thread.